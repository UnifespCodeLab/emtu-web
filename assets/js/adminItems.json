[
  { "title": "Buscas realizadas", "icon": "", "route": "/admin/buscas-realizadas" },
  { "title": "Acessos diários", "icon": "", "route": "/admin/acessos-diarios" },
  { "title": "Ranking de Linhas", "icon": "", "route": "/admin/ranking-linhas" },
  { "title": "Solicitações", "icon": "", "route": "/admin/solicitacoes" }
]