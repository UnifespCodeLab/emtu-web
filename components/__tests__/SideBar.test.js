import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import Vuetify from 'vuetify'

import Sidebar from '~/components/SideBar.vue'
import adminItems from '~/assets/js/adminItems'

describe('Components / Sidebar', () => {
  const localVue = createLocalVue()
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  describe('when component renders', () => {
    beforeEach(() => {
      wrapper = mount(Sidebar, {
        localVue,
        vuetify,
        stubs: {
          'router-link': RouterLinkStub
        },
        mocks: {
          $route: { path: '' }
        },
        propsData: {
          isOpen: true
        }
      })
    })

    describe('when is desktop', () => {
      beforeAll(() => {
        Object.defineProperty(window, 'innerWidth', {
          writable: true, configurable: true, value: 1200
        })
      })

      it('should render 300', () => {
        const navigation = wrapper.findComponent({ name: 'v-navigation-drawer' })
        expect(navigation.props('width')).toBe('300')
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('when is mobile', () => {
      beforeAll(() => {
        Object.defineProperty(window, 'innerWidth', {
          writable: true, configurable: true, value: 500
        })
      })

      it('should render 100%', () => {
        const navigation = wrapper.findComponent({ name: 'v-navigation-drawer' })
        expect(navigation.props('width')).toBe('100%')
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })

  describe('when is not logged', () => {
    it('should render default items', () => {
      wrapper = mount(Sidebar, {
        localVue,
        vuetify,
        stubs: {
          'router-link': RouterLinkStub
        },
        mocks: {
          $route: { path: '/' }
        },
        propsData: {
          isOpen: true
        }
      })

      expect(wrapper.vm.routerItems).toEqual([
        { title: 'Buscar', icon: 'mdi-magnify', route: '/' },
        { title: "Lista de CID's", icon: 'mdi-format-list-numbered', route: '/lista-cids' },
        { title: 'Não encontrei minha rota', icon: 'mdi-bus-alert', route: '/solicitacao' },
        { title: 'Sobre nós', icon: 'mdi-account-group', route: '/sobre-nos' }
      ])
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('when user is logged', () => {
    it('should render admin items', () => {
      wrapper = mount(Sidebar, {
        localVue,
        vuetify,
        stubs: {
          'router-link': RouterLinkStub
        },
        mocks: {
          $route: { path: '/admin' }
        },
        propsData: {
          isOpen: true,
          isUserLogged: true
        }
      })

      expect(wrapper.vm.routerItems).toEqual(adminItems)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
