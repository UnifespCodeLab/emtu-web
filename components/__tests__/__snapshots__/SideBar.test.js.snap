// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Sidebar when component renders when is desktop should render 300 1`] = `
<div
  class="v-sheet theme--light"
>
  <div
    class="v-overlay v-overlay--absolute theme--dark"
    style="z-index: 5;"
  >
    <div
      class="v-overlay__scrim"
      style="opacity: 0; background-color: rgb(33, 33, 33); border-color: #212121;"
    />
  </div>
  <aside
    class="v-navigation-drawer v-navigation-drawer--absolute v-navigation-drawer--is-mobile v-navigation-drawer--open v-navigation-drawer--right theme--light"
    style="height: 100%; top: 0px; transform: translateX(0%); width: 300px; z-index: 1000;"
  >
    <div
      class="v-navigation-drawer__content"
    >
      <div
        class="title-container"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate title-navigation pr-2 mdi mdi-bus theme--light"
          style="color: rgb(1, 123, 253); caret-color: #017BFD;"
        />
         
        <div
          class="v-list-item__title title-navigation"
        >
          
        VApt
      
        </div>
         
        <button
          class="v-icon notranslate title-navigation close-icon v-icon--link mdi mdi-close theme--light"
          style="color: rgb(29, 29, 31); caret-color: #1D1D1F;"
          type="button"
        />
      </div>
       
      <div
        class="v-list v-sheet theme--light v-list--dense v-list--nav"
        role="list"
      >
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-magnify theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Buscar
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-format-list-numbered theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Lista de CID's
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-bus-alert theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              N찾o encontrei minha rota
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-account-group theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Sobre n처s
            </span>
          </div>
        </a>
      </div>
    </div>
    <div
      class="v-navigation-drawer__border"
    />
  </aside>
</div>
`;

exports[`Components / Sidebar when component renders when is mobile should render 100% 1`] = `
<div
  class="v-sheet theme--light"
>
  <div
    class="v-overlay v-overlay--absolute theme--dark"
    style="z-index: 5;"
  >
    <div
      class="v-overlay__scrim"
      style="opacity: 0; background-color: rgb(33, 33, 33); border-color: #212121;"
    />
  </div>
  <aside
    class="v-navigation-drawer v-navigation-drawer--absolute v-navigation-drawer--is-mobile v-navigation-drawer--open v-navigation-drawer--right theme--light"
    style="height: 100%; top: 0px; transform: translateX(0%); width: 100%; z-index: 1000;"
  >
    <div
      class="v-navigation-drawer__content"
    >
      <div
        class="title-container"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate title-navigation pr-2 mdi mdi-bus theme--light"
          style="color: rgb(1, 123, 253); caret-color: #017BFD;"
        />
         
        <div
          class="v-list-item__title title-navigation"
        >
          
        VApt
      
        </div>
         
        <button
          class="v-icon notranslate title-navigation close-icon v-icon--link mdi mdi-close theme--light"
          style="color: rgb(29, 29, 31); caret-color: #1D1D1F;"
          type="button"
        />
      </div>
       
      <div
        class="v-list v-sheet theme--light v-list--dense v-list--nav"
        role="list"
      >
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-magnify theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Buscar
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-format-list-numbered theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Lista de CID's
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-bus-alert theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              N찾o encontrei minha rota
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-account-group theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Sobre n처s
            </span>
          </div>
        </a>
      </div>
    </div>
    <div
      class="v-navigation-drawer__border"
    />
  </aside>
</div>
`;

exports[`Components / Sidebar when is not logged should render default items 1`] = `
<div
  class="v-sheet theme--light"
>
  <aside
    class="v-navigation-drawer v-navigation-drawer--absolute v-navigation-drawer--is-mobile v-navigation-drawer--open v-navigation-drawer--right theme--light"
    style="height: 100%; top: 0px; transform: translateX(0%); width: 100%; z-index: 1000;"
  >
    <div
      class="v-navigation-drawer__content"
    >
      <div
        class="title-container"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate title-navigation pr-2 mdi mdi-bus theme--light"
          style="color: rgb(1, 123, 253); caret-color: #017BFD;"
        />
         
        <div
          class="v-list-item__title title-navigation"
        >
          
        VApt
      
        </div>
         
        <button
          class="v-icon notranslate title-navigation close-icon v-icon--link mdi mdi-close theme--light"
          style="color: rgb(29, 29, 31); caret-color: #1D1D1F;"
          type="button"
        />
      </div>
       
      <div
        class="v-list v-sheet theme--light v-list--dense v-list--nav"
        role="list"
      >
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-magnify theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Buscar
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-format-list-numbered theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Lista de CID's
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-bus-alert theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              N찾o encontrei minha rota
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-account-group theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Sobre n처s
            </span>
          </div>
        </a>
      </div>
    </div>
    <div
      class="v-navigation-drawer__border"
    />
  </aside>
</div>
`;

exports[`Components / Sidebar when user is logged should render admin items 1`] = `
<div
  class="v-sheet theme--light"
>
  <aside
    class="v-navigation-drawer v-navigation-drawer--absolute v-navigation-drawer--is-mobile v-navigation-drawer--open v-navigation-drawer--right theme--light"
    style="height: 100%; top: 0px; transform: translateX(0%); width: 100%; z-index: 1000;"
  >
    <div
      class="v-navigation-drawer__content"
    >
      <div
        class="title-container"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate title-navigation pr-2 mdi mdi-bus theme--light"
          style="color: rgb(1, 123, 253); caret-color: #017BFD;"
        />
         
        <div
          class="v-list-item__title title-navigation"
        >
          
        VApt
      
        </div>
         
        <button
          class="v-icon notranslate title-navigation close-icon v-icon--link mdi mdi-close theme--light"
          style="color: rgb(29, 29, 31); caret-color: #1D1D1F;"
          type="button"
        />
      </div>
       
      <div
        class="v-list v-sheet theme--light v-list--dense v-list--nav"
        role="list"
      >
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate material-icons theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Home
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate material-icons theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Buscas realizadas
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate material-icons theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Acessos di찼rios
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate material-icons theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Ranking de Linhas
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate material-icons theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Solicita챌천es
            </span>
          </div>
        </a>
        <a
          class="v-list-item v-list-item--link theme--light"
          role="listitem"
          router=""
          tabindex="0"
        >
          <div
            class="v-list-item__icon"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate material-icons theme--light"
            />
          </div>
           
          <div
            class="v-list-item__content"
          >
            <span
              class="item-title"
            >
              Fazer uma busca
            </span>
          </div>
        </a>
      </div>
    </div>
    <div
      class="v-navigation-drawer__border"
    />
  </aside>
</div>
`;
